Lib/test/test_email/test_message.py

```diff
@@ -500,6 +500,29 @@ def set_content(self, msg, *args, **kw):
             self.args = args
             self.kw = kw
 
+    def test_add_valid_header(self):
+        m = self._str_msg('')
+        header_name = 'CUSTOM-HEADER'
+        header_value = 'data'
+        m.add_header(header_name, header_value)
+        self.assertEqual(m.get(header_name), header_value)
+
+    def test_throws_on_add_invalid_header_name(self):
+        m = self._str_msg('')
+        header_name = 'CUSTOM\nHEADER'
+        header_value = 'data'
+
+        with self.assertRaises(ValueError):
+            m.add_header(header_name, header_value)
+
+    def test_throws_on_add_invalid_header_value(self):
```

Comment:
maybe add testcase for value continuation (see [content-type/disposition](https://docs.microsoft.com/en-us/previous-versions/office/developer/exchange-server-2010/aa563375(v=exchg.140)))